cub/device/dispatch/dispatch_histogram.cuh

```diff
@@ -1069,12 +1113,17 @@ public:
 
             for (int channel = 0; channel < NUM_ACTIVE_CHANNELS; ++channel)
             {
-                privatized_decode_op[channel].Init(num_output_levels[channel],
-                                                   upper_level[channel],
-                                                   lower_level[channel]);
+              cudaError_t error = privatized_decode_op[channel].Init(num_output_levels[channel],
+                                                                     upper_level[channel],
+                                                                     lower_level[channel]);
 
-                if (num_output_levels[channel] > max_levels)
-                    max_levels = num_output_levels[channel];
+              if (error != cudaSuccess)
+              {
+                return error;
```

Comment:
I believe that you return the error earlier than setting `temp_storage_bytes`. If users accidentally ignore return code, they risk allocating temporary storage of a random size. This random size is usually larger than available memory. In any way, the contract is that we set temporary storage in the first invocation of the API, so the users are allowed to avoid initializing it. Let's set `temp_storage_bytes` to something meaningful here just in case (probably `0`).

Reply:
Good point 👍  Wasn't very sure about our guarantees. But I can totally see how users would not check the return value when querying memory requirements.